### PR TITLE
fix: GCC13 compile failed

### PR DESCRIPTION
**Pull request by 4kangjc**

### CODE REVIEW

**Comment by 4kangjc** [comment]
commit 4 see https://gcc.gnu.org/gcc-13/porting_to.html.

**Comment by 0x804d8000** [comment]
Merged. Thanks for the patch!

**Comment by 4kangjc** [comment]
https://github.com/Tencent/flare/blob/bf9b82edc217cc9cafdb0336d6cfb85209d90dfd/flare/base/thread/latch.cc#L23-L30
这里可以优化一下? notify不用在锁里面来着.
```cpp
void Latch::count_down(std::ptrdiff_t update) {
  bool count_is_zero = false;
  {
    std::lock_guard _(m_);
    FLARE_CHECK_GE(count_, update);
    count_is_zero = !(count_ -= update);
  }
  if (count_is_zero) {
    cv_.notify_all();
  }
}
```

**Comment by 0x804d8000** [comment]
嗯，这个应该可以优化

**Comment by 4kangjc** [comment]
> 嗯，这个应该可以优化

OK, 我提一条PR :)

File: flare/base/maybe_owning_test.cc [hunk 1]
```diff
@@ -161,9 +161,16 @@ TEST(MaybeOwning, SelfMove) {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wself-move"
 #endif
+#if __GNUC__ >= 13
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wself-move"
     ppp = std::move(ppp);
+#endif
```
Comment (4kangjc) [inline flare/base/maybe_owning_test.cc:168]: my fault. `#endif` should be above `ppp = std::move(ppp);`

**Pull request merged**